test/benchmark/BenchmarkBigIT.java

```diff
@@ -0,0 +1,395 @@
+                        queryPattern +
+                        " $y iid " + entityId.getIID().toHexString() + ";";
+                String limitedQueryString = "match " + queryPattern +
+                        " limit " + limit + ";";
+
+                executeQuery(queryString, tx, "full");
+                executeQuery(subbedQueryString, tx, "first argument bound");
+                executeQuery(subbedQueryString2, tx, "second argument bound");
+                executeQuery(limitedQueryString, tx, "limit " + limit);
+            }
+        }
+    }
+
+    /**
+     * Scalability test defined in terms of number of rules in the system. Creates a simple rule chain based on join operation on two relations:
+     * <p>
+     * R_i(x, y) := R_1(x, z), R_{i-1}(z, y);    i e [2, N],
+     * <p>
+     * initialised with a chain of N x base relation R_1instances.
+     * <p>
+     * The rules are defined in such a way that relation R_j spans j hops of base relations and has (N + 1 - j) instances.
+     * Consequently R_N has a single instance linking the first and last entities in the base relation chain.
+     */
+    @Test
+    public void testJoinRuleChain() {
+        final int N = 100;  // Stack-size of 200 causes an overflow
+        System.out.println(new Object() {
+        }.getClass().getEnclosingMethod().getName());
+        loadRuleChainData(N);
+
+        try (TypeDB.Session session = dataSession()) {
+            try (TypeDB.Transaction tx = session.transaction(Arguments.Transaction.Type.READ, new Options.Transaction().infer(true))) {
+                Thing firstId = tx.query().match(TypeQL.parseQuery("match $x has index 'first';").asMatch()).next().get("x").asThing();
+                Thing lastId = tx.query().match(TypeQL.parseQuery("match $x has index '" + N + "';").asMatch()).next().get("x").asThing();
+                String queryPattern = "(fromRole: $x, toRole: $y) isa relation" + N + ";";
+                String queryString = "match " + queryPattern;
+                String subbedQueryString = "match " + queryPattern +
+                        "$x iid " + firstId.getIID().toHexString() + ";";
+                String subbedQueryString2 = "match " + queryPattern +
+                        "$y iid " + lastId.getIID().toHexString() + ";";
+                String limitedQueryString = "match " + queryPattern +
+                        " limit 1;";
+                assertEquals(1, executeQuery(queryString, tx, "full").size());
+                assertEquals(1, executeQuery(subbedQueryString, tx, "first argument bound").size());
+                assertEquals(1, executeQuery(subbedQueryString2, tx, "second argument bound").size());
+                assertEquals(1, executeQuery(limitedQueryString, tx, "limit ").size());
+            }
+        }
+    }
+
+    @Test
+    public void whenQueryingConcurrently_reasonerDoesNotThrowException() throws ExecutionException, InterruptedException {
```

Comment:
i mean... im not sure we need this test anymore if it only is here for what the title says

Reply:
Will remove